### PR TITLE
Additional CP calculations

### PR DESCRIPTION
**Pull request by pvanassen**
Added getting the max attainable cp for a fully evolved and powerup'd version of this pokemon
And made the utils class public for easier testing

### CODE REVIEW

**Comment by Jari27** [comment]
This already exists right? Pokemon.getMaxCp(PokemonId id)

**Comment by pvanassen** [comment]
Can't find it anywhere. Do you have a file + line number for me?

**Comment by pvanassen** [comment]
Oh wait, that's been moved to PokemonDetails. Was on an older version. 
Moving stuff

**Comment by Grover-c13** [comment]
Pokemon inherits PokemonDetails so Pokemon.getMaxCp still works

**Comment by pvanassen** [comment]
Pokemon.getMaxCp(PokemonId id) will not take into account the current values of the current pokemon. This is a different method giving you the CP if you evolve this pokemon and powerup it to the max.

**Comment by pvanassen** [comment]
More bits are missing

**Comment by pvanassen** [comment]
While testing and checking I found some weird issues. Investigating

**Comment by stirante** [comment]
Maybe add also a way to get how much cp will pokemon have after just evolving?

**Comment by pvanassen** [comment]
That's not too hard to do. Will take a look.

**Comment by pvanassen** [comment]
Fixed a bunch of calculations based on field testing
Added methods for getting max cp based on the player's level
Added CP after evolution

**Comment by pvanassen** [comment]
Can be merged

File: library/src/main/java/com/pokegoapi/api/pokemon/PokemonDetails.java [hunk 1]
```diff
@@ -234,7 +266,68 @@ public int getAbsoluteMaxCp() throws NoSuchItemException {
...
+	}
 
+	/**
+	 * Calculated the max cp of this pokemon, if you upgrade it fully with your current player level
+	 * @return Max cp of this pokemon
+	 */
+	private int getMaxCpFullEvolveAndPowerup(int playerLevel) {
+		PokemonIdOuterClass.PokemonId highestUpgradedFamily;
+		if (getPokemonId() == EEVEE) {
+			highestUpgradedFamily = PokemonIdOuterClass.PokemonId.FLAREON;
```
Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/PokemonDetails.java]: static imports for the other 4 but not for this?

Comment (pvanassen) [inline library/src/main/java/com/pokegoapi/api/pokemon/PokemonDetails.java]: IDE magic?

**Comment by FabianTerhorst** [comment]
👍

**Pull request merged**